pkg/crane/append.go

```diff
@@ -90,7 +107,11 @@ func getLayer(path string, layerType types.MediaType) (v1.Layer, error) {
 		return stream.NewLayer(f, stream.WithMediaType(layerType)), nil
 	}
 
-	return tarball.LayerFromFile(path, tarball.WithMediaType(layerType))
+	if layerType == types.OCILayerZStd {
+		return tarball.LayerFromFile(path, tarball.WithMediaType(layerType), tarball.WithCompression(compression.ZStd))
```

Comment:
I don't think this works properly. Imagine you gave it a `.tar.gz` file and asked for zstd compression. No re-compression will happen. You will have zstd in manifest, but contents of the blob will be gzip.

Note that there is a bug even before this PR: if one calls `crane append` with a zstd file, it will blindly append it with gzip in manifest.